File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnClassQueryFromSense.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.verbnet.sql;

import android.database.sqlite.SQLiteDatabase;

import org.sqlunet.sql.DBQuery;

/**
 * VerbNet class-with-sense query
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
class VnClassQueryFromSense extends DBQuery
{
	/**
	 * <code>QUERY</code> is the SQL statement
	 */
	static private final String QUERY = SqLiteDialect.VerbNetClassQueryFromSense;

	/**
	 * Constructor
	 *
	 * @param connection connection
	 * @param wordId     target word id
	 * @param synsetId   target synset id (null corresponds to no value)
	 */
	@SuppressWarnings("boxing")
	public VnClassQueryFromSense(final SQLiteDatabase connection, final long wordId, final Long synsetId)
	{
		super(connection, VnClassQueryFromSense.QUERY);
		setParams(wordId, synsetId);
	}

	/**
	 * Get the class id from the result set
	 *
	 * @return the class id from the result set
	 */
	public long getClassId()
	{
		assert this.cursor != null;
		return this.cursor.getLong(0);
	}

	/**
	 * Get the class name from the result set
	 *
	 * @return the class name from the result set
	 */
	public String getClassName()
	{
		assert this.cursor != null;
		return this.cursor.getString(1);
	}

	/**
	 * Get the frame xtag from the result set
	 *
	 * @return the frame xtag from the result set
	 */
	public boolean getSynsetSpecific()
	{
		assert this.cursor != null;
		return this.cursor.getInt(2) == 0;
	}

	/**
	 * Get the definition from the result set
	 *
	 * @return the definition from the result set
	 */
	public String getDefinition()
	{
		assert this.cursor != null;
		return this.cursor.getString(3);
	}

	/**
	 * Get the sensenum from the result set
	 *
	 * @return the sensenum from the result set
	 */
	public int getSenseNum()
	{
		assert this.cursor != null;
		return this.cursor.getInt(4);
	}

	/**
	 * Get the sensekey from the result set
	 *
	 * @return the sensekey from the result set
	 */
	public String getSenseKey()
	{
		assert this.cursor != null;
		return this.cursor.getString(5);
	}

	/**
	 * Get the quality from the result set
	 *
	 * @return the quality from the result set
	 */
	public float getQuality()
	{
		assert this.cursor != null;
		return this.cursor.getFloat(6);
	}

	/**
	 * Get the groupings from the result set
	 *
	 * @return the (|-separated) groupings from the result set
	 */
	public String getGroupings()
	{
		assert this.cursor != null;
		return this.cursor.getString(7);
	}
}

File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnClassQueryFromSense.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.verbnet.sql

import android.database.sqlite.SQLiteDatabase
import org.sqlunet.sql.DBQuery

/**
 * VerbNet class-with-sense query
 *
 * @param connection connection
 * @param wordId     target word id
 * @param synsetId   target synset id (null corresponds to no value)
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
internal class VnClassQueryFromSense(connection: SQLiteDatabase, wordId: Long, synsetId: Long?) : DBQuery(connection, QUERY) {

    init {
        setParams(wordId, synsetId!!)
    }

    /**
     * Class id from the result set
     */
    val classId: Long
        get() = cursor!!.getLong(0)

    /**
     * Class name from the result set
     */
    val className: String
        get() = cursor!!.getString(1)

    /**
     * Frame xtag from the result set
     */
    val synsetSpecific: Boolean
        get() = cursor!!.getInt(2) == 0

    /**
     * Definition from the result set
     */
    val definition: String
        get() = cursor!!.getString(3)

    /**
     * Sensenum from the result set
     */
    val senseNum: Int
        get() = cursor!!.getInt(4)

    /**
     * Sensekey from the result set
     */
    val senseKey: String
        get() = cursor!!.getString(5)

    /**
     * quality from the result set
     */
    val quality: Float
        get() = cursor!!.getFloat(6)

    /**
     * (|-separated) Groupings from the result set
     */
    val groupings: String
        get() = cursor!!.getString(7)


    companion object {
        /**
         * `QUERY` is the SQL statement
         */
        private const val QUERY = SqLiteDialect.VerbNetClassQueryFromSense
    }
}
